### PR TITLE
Allow pass network name as kwarg for streams

### PR DESCRIPTION
**Pull request by FranzForstmayr**
Solves #594 finally.

### CODE REVIEW

File: skrf/network.py [hunk 1]
```diff
@@ -429,6 +429,8 @@ def __init__(self, file: str = None, name: str = None, params: dict = None,
 
             # allows user to pass StringIO, filename or file obj
             if isinstance(file, io.StringIO):
+                if not hasattr(file, "name") and name is not None:
+                    file.name = name
```
Comment (Asachoo) [inline skrf/network.py:433]: Is it essential to verify the format of the filename and ensure it conforms to the pattern `r"*.[ghsyz](\d+)p"`?

Comment (FranzForstmayr) [reply skrf/network.py:433]: The file name is verified inside the `Touchstone._parse_file` https://github.com/scikit-rf/scikit-rf/blob/master/skrf/io/touchstone.py#L375 function which is called afterwards, so here no verification necessary.

**Review by Asachoo — APPROVED** [review]

**Comment by Asachoo** [comment]
I've noticed that the `name` property in the `Network` class can be a bit inconsistent. Let me illustrate my point with an example:
```python3
(scikit-rf) ➜  scikit-rf git:(master) ✗ ipython
Python 3.13.0 (main, Oct 16 2024, 08:05:40) [Clang 18.1.8 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.29.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import skrf as rf; from io import StringIO
matplotlib not found while setting up plotting

In [2]: ntwk1 = rf.Network('./skrf/tests/ntwk1.s2p'); ntwk1
Out[2]: 2-Port Network: 'ntwk1',  1.0-10.0 GHz, 91 pts, z0=[50.+0.j 50.+0.j]

In [3]: with open('./skrf/tests/ntwk1.s2p') as f:
   ...:     sio = StringIO(f.read())
   ...:     ntwk2 = rf.Network(sio, name='./skrf/tests/ntwk1.s2p')
   ...: ntwk2
Out[3]: 2-Port Network: './skrf/tests/ntwk1.s2p',  1.0-10.0 GHz, 91 pts, z0=[50.+0.j 50.+0.j]

In [4]: with open('./skrf/tests/ntwk1.s2p') as f:
   ...:     sio = StringIO(f.read())
   ...:     ntwk3 = rf.Network(sio, name='ntwk1.s2p')
   ...: ntwk3
Out[4]: 2-Port Network: 'ntwk1.s2p',  1.0-10.0 GHz, 91 pts, z0=[50.+0.j 50.+0.j]

In [5]: ntwk4 = rf.Network(s=ntwk1.s, frequency=ntwk1.frequency, name='ntwk4'); ntwk4
Out[5]: 2-Port Network: 'ntwk4',  1.0-10.0 GHz, 91 pts, z0=[50.+0.j 50.+0.j]

In [6]: ntwk5 = rf.Network('./skrf/tests/ntwk1.s2p', name='ntwk1.s2p'); ntwk5
Out[6]: 2-Port Network: 'ntwk1.s2p',  1.0-10.0 GHz, 91 pts, z0=[50.+0.j 50.+0.j]

In [7]: 
```

As you can see, when creating Network objects using "conventional" methods like `ntwk1` and `ntwk4`, the name property omit the `.s2p` extension.

However, for `Network` objects created with `StringIO`, such as `ntwk2` and `ntwk3`, still cling to the `extension` even when the path is redundant.

While there's no strict rule on the format of the Network's name, the sporadic inclusion of the extension can feel a bit inconsistent.

I understand that this is not a bug, but perhaps automatically removing the extension when parsing with `StringIO` would make the naming more consistent. Just brainstorming here, not advocating for a change :laughing:.

**Comment by FranzForstmayr** [comment]
I see your point, and now I cannot unsee it anymore. :)

we need the proper extension for the parser, but we can remove it afterwards.

**Comment by Ttl** [comment]
This has a side effect of adding `name` attribute to the user input. It likely isn't a big issue but it would be better to not modify the user input.

**Comment by FranzForstmayr** [comment]
Name was already a user input, even before this PR. In this case the name is just used if the file argument is a stream.

**Comment by FranzForstmayr** [comment]
Yes that's true, I also think no big deal. There's definitely no breaking change, as this code raised an exception before this change.

**Pull request merged**